### ISSUE TITLE
Binary data should be represented by `Uint8List`s

### ISSUE THREAD

**Issue by ueman** [brendan-duncan/archive#355]
Using `Uint8List` is more performant and resourceful for representing binary data.
The archive package should therefore try to use `Uint8List`s instead of `List<int>`s.

See also the notes on its documentation [here](https://api.dart.dev/stable/3.5.3/dart-typed_data/Uint8List-class.html).

**Comment by brendan-duncan** [comment]
Yes, the library does already use Uint8List internally. List<int> is the interface Uint8List implements.

**Comment by ueman** [comment]
It would be nice to then also expose it as such for v4. Having `List<int>` as a type to work with makes glue code annoying, since you always have to check whether a `List<int>` is a `Uint8List` or you have to convert it to a `Uint8List`.

Also, thanks for providing this awesome library!

**Comment by brendan-duncan** [comment]
I've been away from working on this library for a while, but am getting back to it to get V4 ready to ship. 

Taking a look at V4 it should already be using Uint8List for return values, except for some alias methods I kept in for backward compatibility. They'll generally still use List<int> for arguments, to be more flexible.
